### ISSUE TITLE
[15.3 feature] RestoreAdditionalSources, if set, will prevent searching normal sources from config

### ISSUE THREAD

**Issue by natidea** [NuGet/Home#5478]
## Repro steps

1. Installing VS on a clean VM and install the dotnet CLI (2.0.0-preview3-006546). This leaves behind an empty folder under C:\Program Files\dotnet\sdk\NuGetFallbackFolder because of https://github.com/dotnet/cli/issues/6989

2. In VS create a new netcoreapp2.0 project (restores successfully), and change the target framework to netcoreapp1.1

**Expected**
Restores successfully

**Actual**
Restore fails with error:
Error	NU1101	Unable to find package Microsoft.NETCore.App. No packages exist with this id in source(s): C:\Program Files\dotnet\sdk\NuGetFallbackFolder

**Workaround**
Deleting C:\Program Files\dotnet\sdk\NuGetFallbackFolder allows NuGet to restore successfully	

## Details

`dotnet --info` output:

.NET Command Line Tools (2.0.0-preview3-006546)

Product Information:
 Version:            2.0.0-preview3-006546
 Commit SHA-1 hash:  f4d884a03d

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.0-preview3-006546\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0-preview3-25419-01
  Build    : 2fab74ca0ab629adfe1291e3f1b5eeae2a57c156

**Comment by MiaomiaoHuangfu** [comment]
Install VS15.3 preview 3(d15Prerel.26627.3003) + CLI/SDK 2.0 preview 3 (2.0.0-preview3-006607);
 
Create a new .NET Core project (C# or VB );
Right click the project and click Properties;
- In Application tab:
- If changing the Target framework from ".NET Core 2.0" to ".NET Core 1.1", restore failed with an error in error list window "NU1103: Unable to find a stable package Microsoft.NETCore.App with version (>= 1.1.2)- Found 1 version(s) in C:\Program Files\dotnet\sdk\NuGetFallbackFolder [ Nearest version: 2.0.0-preview2-25407-01 ]";
- If changing the TFM to ".net core 1.0", restore succeeded.

**dotnet --info**
```
.NET Command Line Tools (2.0.0-preview3-006607)

Product Information:
 Version:            2.0.0-preview3-006607
 Commit SHA-1 hash:  81dd304f72

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.0-preview3-006607\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0-preview3-25426-01
  Build    : 122309e7e7fb9bb233fe4910f822c28ec73f0957
```

**Comment by natidea** [comment]
Note: when I changed to netcoreapp1.0, it also failed to restore. So I assume it worked for Miaomiao because she had the 1.0 bits already restored somewhere. 

As with the empty NuGetFallbackFolder case, deleting `C:\Program Files\dotnet\sdk\NuGetFallbackFolder` allows the netcoreapp1.1 project to restore successfully.

**Comment by mishra14** [comment]
@zhili1208 who is on hotseat.

**Comment by nkolev92** [comment]
I think I've uncovered the root of the problem but it doesn't explain all the scenarios described above. 

The issue happens when the RestoreAdditionalProjectSources

More NuGet info here:
https://github.com/NuGet/Home/wiki/%5BSpec%5D-NuGet-settings-in-MSBuild
The reasoning for this property is that we want to vary the fallbackfolder as a source in netcore1.x projects, and as a fallback folder in netcore2.x projects
https://github.com/dotnet/sdk/pull/1323

The SDK (version 2.0.0-preview3-006611) passes this value in the nomination to NuGet.
The value by the SDK is set as: 
```
C:\Program Files\dotnet\sdk\2.0.0-preview3-006611\Sdks\Microsoft.NET.Sdk\build\..\..\..\..\NuGetFallbackFolder
```
So what happens here is that restore sources is supposed to take precedence over the sources in the config as described in the spec.

RestoreAdditionalProjectSources is not a property that NuGet uses, so we combine the RestoreSources and RestoreAdditionalProjectSources into 1 value.
The combining is done during nomination, so when we restore we think that RestoreSources were specified so we don't read the configs. 

**tldr;** 
RestoreSources being empty & RestoreAdditionalProjectSources having a value don't play well together.

The impact here is that projects with SDK 2.0 targetting netcore1.x framework will not use the correct sources, hence users will not be able to install any nuget packages. \


@natidea 
```
As with the empty NuGetFallbackFolder case, deleting C:\Program Files\dotnet\sdk\NuGetFallbackFolder allows the netcoreapp1.1 project to restore successfully.
```
I am still not sure how to explain this however, since the fallback folder is deleted and it is specified as a source, restore should fail fast. 
https://github.com/NuGet/Home/issues/3058

**Note:** 
This is not an issue with CLI. 
It's only an issue in VS restore. 

//cc 
@livarcocc @nguerrera @rrelyea

**Comment by livarcocc** [comment]
I believe in @natidea's case, it is not that the folder has been deleted, it is that it is empty and NuGet is failing because it is empty. When he deletes it, we stop adding it as a source/fallback and NuGet succeeds.

**Comment by nkolev92** [comment]
That makes sense.
@livarcocc when did the fallback folder start being added as a source in for netcore 1.x?
As far as I can tell, it's only happening in 2.0.0 preview3.

**Comment by livarcocc** [comment]
A few weeks ago, for preview3. That was the plan to support fallback folder and a global offline cache.

**Comment by nkolev92** [comment]
Related to this issue:
https://github.com/NuGet/Home/issues/5529
